### PR TITLE
Replaced ids with uuids

### PR DESCRIPTION
**Pull request by farhatahmad**
## Description

Sequential ids are slightly problematic
1- They're guessable (ie add 1 to get to the next user)
2- They provide extra unneeded information (ie how many users)

I replaced them with `uuids` in the database

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def create
         return render_error errors: user.errors.to_a if hcaptcha_enabled? && !verify_hcaptcha(response: params[:token])
 
         if user.save
-          session[:user_id] ||= user.id
+          session[:user_id] ||= user.reload.id
```
Comment (scouillard) [inline app/controllers/api/v1/users_controller.rb:33]: This doesn't seem to work.
Upon creating an account, it does not log you in - it buffers until the page crashes.

**Review by KH-Amir-TN — APPROVED** [review]

File: app/controllers/api/v1/users_controller.rb (hunk 1 above)
Comment (farhatahmad) [reply app/controllers/api/v1/users_controller.rb:33]: Hmm it was crashing before I added and the reload fixed it - let me retest

Comment (farhatahmad) [reply app/controllers/api/v1/users_controller.rb]: Just had to do with a stale session - fixed it by removing the `||=`

**Comment by scouillard** [comment]
We need the ||= because we are using the users#create endpoint as an admin to create a new user.

We don't want to replace the session[:user_id] with the new user if the new user is made via the admin interface.

**Review by KH-Amir-TN — APPROVED** [review]

**Pull request merged**